### ISSUE TITLE
Neural Network extract layer activations

### ISSUE THREAD

**Issue by rlk41** [rlk41/laminar_fmri#8]
I’ve mainly been working to use neural networks to generate labels/extract features for each frame. The idea is to use these features to build dissimilarity matrices to correlate with the neural dissimilarity matrices (barcodes)--see where this information is being represented. I’ve tried some pre-trained neural networks (resnet, VGG16) and looking at the activations at intermediate layers. These neural networks don’t necessarily build biologically plausible intermediate representations, so might want to try some different models like HMAX. I’ve tried a couple of unsupervised methods for segmenting the frames (background, foreground, objects) but they didn’t work too well, so I’ve been playing around with a more constrained approach--although with a little more time I could probably get the unsupervised methods to work. 

Mask-RCNN (blog post) can be trained to generate masks for each frame, extract pixel data using mask, and push data through a neural network. I’ve been annotating some data using the attributes--character name, character orientation (right,left,back,front) and “polyline” outline (x,y coordinates) of the character. I was thinking we could use this data to build dissimilarity matrices based on face features. Not clear how the brain encodes faces so there is some variability here, but you could imagine extracting face features like eyes, size of eyes, eyebrows, nose, mouth, etc… and build a dissimilarity matrix based on these. One other paper does something similar ([Comparison of deep neural networks to spatio-temporal cortical dynamics of human visual object recognition reveals hierarchical correspondence](https://dspace.mit.edu/handle/1721.1/103585) ) but they use whole scenes and presumably this limits them to V1-V5--the only place whole scene information can be within one searchlight. If we break the images down we might be able to get into more specialized areas like the FFA, VWFA or beyond V5. 

**I think the main difference between this and the MIT paper is that the MIT paper uses layer activations corresponding to the entire visual field to perform RSA. It might be interesting to get activations for only faces and performs RSA. The mask-rrn might enable us to do that**

![image](https://user-images.githubusercontent.com/20213755/123305321-f81e3900-d4ed-11eb-932b-90948d6b5e4c.png)
![image](https://user-images.githubusercontent.com/20213755/123305375-08ceaf00-d4ee-11eb-911c-d16007d9602d.png)
![image](https://user-images.githubusercontent.com/20213755/123305406-1126ea00-d4ee-11eb-9865-05c699082627.png)
![image](https://user-images.githubusercontent.com/20213755/123305434-184df800-d4ee-11eb-96cc-8e3b23a424dd.png)
![image](https://user-images.githubusercontent.com/20213755/123305455-1d12ac00-d4ee-11eb-91ab-b57de586da8f.png)
